library/analytics/src/test/java/org/cru/godtools/analytics/firebase/FirebaseAnalyticsServiceTest.kt

```diff
@@ -43,7 +43,7 @@ class FirebaseAnalyticsServiceTest {
         eventBus = mock()
         firebase = mock()
         oktaUserProfileProvider = mock {
-            on { userInfoFlow() } doReturn userInfoChannel.consumeAsFlow()
+            on { userInfoFlow(refreshIfStale = any()) } doReturn userInfoChannel.consumeAsFlow()
```

Comment:
Ah, this makes sense, nice catch!

Reply:
yeah, would have been nice if the test had failed instead of just hung. one drawback to testing coroutines is that a lot of time you are waiting for code to execute in the background.

Reply:
in this case without the proper mock statement the test channel was never attached to the `userInfoFlow()` within `FirebaseAnalyticsService`. This in turn meant the test values i was pushing into the channel were never delivered and state was never updated.